docusaurus/docs/cms/plugins-development/widgets-api.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Widgets API
+description: Learn how to use the Widgets API to create and manage widgets in the Strapi admin panel.
+displayed_sidebar: cmsSidebar
+tags:
+  - widgets
+  - admin panel
+  - plugins development
+---
+
+# Widgets API
+
+The Widgets API allows you to create and manage widgets in the Strapi admin panel. Widgets are customizable components that can be added to various parts of the admin interface to display information or provide additional functionality.
+
+## Creating a Widget
+
+To create a widget, you need to use the `Widgets` class from the `@strapi/admin` package. Here's an example of how to create a widget:
+
+```javascript
+import { Widgets } from '@strapi/admin';
+
+const widgets = new Widgets();
+
+widgets.register({
+  id: 'my-widget',
+  pluginId: 'my-plugin',
+  icon: MyWidgetIcon,
+  title: {
+    id: 'my-widget.title',
+    defaultMessage: 'My Widget',
+  },
+  component: () => import('./components/MyWidget'),
+});
+```
+
+The `register` method accepts an object with the following properties:
+
+- `id` (string): A unique identifier for the widget.
+- `pluginId` (string, optional): The ID of the plugin if the widget belongs to a specific plugin.
+- `icon` (React.ComponentType): The icon component to display for the widget.
+- `title` (MessageDescriptor): The title of the widget, using React Intl format.
+- `component` (function): A function that returns a Promise resolving to the widget's React component.
+- `permissions` (array, optional): An array of permission objects to control access to the widget.
+
+## Managing Widgets
+
+The `Widgets` class provides methods to manage widgets:
+
+### getAll()
+
+Retrieves all registered widgets:
+
+```javascript
+const allWidgets = widgets.getAll();
+```
```

Comment:
this isn't useful to users, it just allows us to render the homepage. Registering is all that matters for them so I recommend to omit the rest